crates/neon-runtime/src/napi/lifecycle.rs

```diff
@@ -0,0 +1,39 @@
+use std::mem::MaybeUninit;
```

Comment:
I think it would be helpful for the docs to talk explicitly about these entities:
- "agents" (i.e. the main thread + workers)
- modules
- Roots
and how their lifecycles relate to each other.